File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    for (let i=0, lastNonZeroIdx=0; i < nums.length; i++) {
        if (nums[i] !== 0) {
            [nums[lastNonZeroIdx], nums[i]] = [nums[i], nums[lastNonZeroIdx]]
            lastNonZeroIdx++
        }
    }
    return nums
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        def swap(i: int, j: int) -> None:
            nonlocal nums
            temp = nums[i]
            nums[i] = nums[j]
            nums[j] = temp
        
        last_non_zero_idx = 0
        for i, n in enumerate(nums):
            if n != 0:
                swap(last_non_zero_idx, i)
                last_non_zero_idx += 1
        